ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetaDataAccessImpl.java

```diff
@@ -0,0 +1,156 @@
+        try {
+            final FileSystem store = file.getFileSystem();
+            final Set<String> sets = store.supportedFileAttributeViews();
+            for (final String set : sets) {
+                switch (set) {
+                    case "basic":
+                        final BasicFileAttributes attr = Files.readAttributes(file, BasicFileAttributes.class);
+                        metadata.put("x-amz-meta-ds3-creationTime", String.valueOf(attr.creationTime().toMillis()));
+                        metadata.put("x-amz-meta-ds3-accessTime", String.valueOf(attr.lastAccessTime().toMillis()));
+                        metadata.put("x-amz-meta-ds3-lastModifiedTime", String.valueOf(attr.lastModifiedTime().toMillis()));
+                        metadata.put("x-amz-meta-ds3-isDirectory", String.valueOf(attr.isDirectory()));
+                        metadata.put("x-amz-meta-ds3-isOther", String.valueOf(attr.isOther()));
+                        metadata.put("x-amz-meta-ds3-isRegularFile", String.valueOf(attr.isRegularFile()));
+                        metadata.put("x-amz-meta-ds3-isSymbolicLink", String.valueOf(attr.isSymbolicLink()));
+                        metadata.put("x-amz-meta-ds3-size", String.valueOf(attr.size()));
+                        String ext = FilenameUtils.getExtension(file.getParent() + "/" + file.getFileName());
+                        metadata.put("x-amz-meta-ds3-fileFormat", ext);
+                        String os = System.getProperty("os.name");
+                        metadata.put("x-amz-meta-ds3-os", os);
+                        //if os is windows then posix will not be called and we need to find permission in different manner
+                        if (os.contains("Windows")) {
+                            saveWindowsfilePermissions(file, metadata);
+                        }
+                        break;
+
+                    case "owner":
+                        final FileOwnerAttributeView ownerAttributeView = Files.getFileAttributeView(file, FileOwnerAttributeView.class);
+                        final UserPrincipal owner = ownerAttributeView.getOwner();
+                        metadata.put("x-amz-meta-ds3-owner", owner.getName());
+                        break;
+
+                    case "user":
+                        break;
+
+                    case "unix":
+                        break;
+
+                    case "posix":
+                        final PosixFileAttributes attr1 = Files.readAttributes(file, PosixFileAttributes.class);
+                        // metadata.put("x-amz-meta-ds3-owner", attr1.owner().getName());
+                        metadata.put("x-amz-meta-ds3-groupName", attr1.group().getName());
+                        metadata.put("x-amz-meta-ds3-permisions", getModes(PosixFilePermissions.toString(attr1.permissions())));
+                        break;
+                }
+            }
+        } catch (final IOException e) {
+            e.printStackTrace();
```

Comment:
Exceptions should not be handled in this way.  At a minimum it should be logged with a slf4j logger, or wrapped in a runtime exception and re-thrown.